### PR TITLE
Create user on first token exchange and implement GET /users/:id

### PR DESCRIPTION
**Pull request by kstam**

### CODE REVIEW

File: src/api/routes/user.test.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+          expect((user.id = id.value())).toEqual(id.value());
+          expect(user.creationTime).toBeDefined();
+          expect(user.email).toBeDefined();
+        });
+    });
+  });
+});
+
+afterAll((done) => {
+  database.destroy().then(done);
```
Comment (Tankenstein) [inline src/api/routes/user.test.ts]: can just `return database.destroy()`, jest is integrated with promises

File: src/api/routes/user.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+  route.get(
+    '/users/:id',
+    async (req: Request, res: Response, next: (...things: any[]) => any) => {
+      const { id } = req.params;
+
+      const user = await getUser.byId(id);
+
+      if (!user) {
+        next(new ApiError(404, 'user.not-found'));
```
Comment (Tankenstein) [inline src/api/routes/user.ts]: I would `throw` this instead

File: src/domain/model/user/User.ts [hunk 3]
```diff
@@ -1,8 +1,12 @@
...
 
 export class User {
-  constructor(private _id: string, private _email: Email) {}
+  constructor(
+    private _id: UserId,
+    private _email: Email,
+    private _creationTime: Date = new Date()
+  ) {}
 
   public id() {
```
Comment (Tankenstein) [inline src/domain/model/user/User.ts]: You could also do:
```
class Thing {
  public readonly email: Email;
}
```
And the type system will enforce immutability

Comment (Tankenstein) [reply src/domain/model/user/User.ts]: You could also use straight-up getters:

```
public get email() {
  return this._email;
}
```

Then you can just do `user.email`

Comment (kstam) [reply src/domain/model/user/User.ts]: hmmm nice.. didn't know of this..

**Pull request merged**